_posts/2015-09-18-Cocoa-Architecture-Dependencies.markdown

```diff
@@ -0,0 +1,129 @@
+
+These can be subtle for example, using Fastlane introduces a lot of dependencies.
+
+``` sh
+~/dev/scratch ⏛  cat Gemfile
+gem 'fastlane'
+
+~/dev/scratch ⏛  bundle install
+[...]
+Bundle complete! 1 Gemfile dependency, 73 gems now installed.
+```
+
+There's a trade-off here, Fastlane provides a great programmable API to a bunch of really annoying time-consuming tasks. There isn't a system similar to CocoaPods' subspecs which lets a library consumer choose to use a subset of a dependency graph, so instead everyone gets `slack-notifier` regardless of if you need it.
+
+ReactiveCocoa feels quite similar. In exchange for Cocoa-native approach to Functional Reactive Programming you also have:
+
+* Swizzling dealloc on objects in order to do it's magic KVO-unbinding at runtime.
+* To pretty much giving up on trying to use the stack trace for understanding flow. Instead you can use their custom dtrace instrument.
+* Complicated pre-compiler macros that can get tricky to debug.
+
+These trade-offs can be happily made in exchange for ReactiveCocoa's well thought out API. Functional Reactive Programming is something that the majority of us are particularly interested in. Since Swift came out, a few Swift-only FRP libraries have been released. So we've been keeping our eyes [on the alternatives](https://cocoapods.org/?q=summary%3Areactive).
+
+#### Focus
+
+Ideally you're bringing in a dependency for one specific task. Something like [ObjectiveSugar](https://cocoapods.org/pods/ObjectiveSugar) is a great example of a small focused library. I studied it's API when we first integrated it, and it's not really changed at all since. The library authors have done a great job of ensuring that Objective Sugar stays on-topic.
+
+#### Community Relationships
+
+The amount of energy you have to put into using certain dependencies, in an engaging way like we do, also means you have to deal with people. This is often the case with OSS, so we’re used to that and quite good at it. Sometimes, however, people and their opinions don’t match, it’s a fact of life, and in these cases it can be worth it to choose to not use a certain dependency, if not strictly needed.
+
+One such example has been ReactiveCocoa. While they are completely in their right to do whatever they feel they need to do, we feel that we’ve wasted energy on supporting their dependencies in CocoaPods that took away from our already constraining time-budget. It consumes and creates energy that we don't want to be associated with, and makes us want to engage less with the community at whole, which is really not something we want. So that, combined with our opinion on the framework itself (as aforementioned) means we’re looking at alternatives.
```

Comment:
> While they are completely in their right to do whatever they feel they need to do

Sounds passive aggressive to my delicate Canadian ears. Maybe "while they're free to make whatever decisions they like..."

> It consumes and creates energy

_I_ know what you mean by "creates energy", but others may not. 

> makes us want to engage less with the community at whole

Explaining why would be useful, I think. Maybe say "their attitude/behaviour/demeanour demotivates us" or something?